core/src/main/java/org/opensearch/sql/calcite/utils/WildcardReplaceUtils.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.calcite.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Utility for wildcard-based string replacement in PPL replace command.
+ *
+ * <p>Supports SPL-style wildcard matching where '*' matches zero or more characters. Captured
+ * wildcard portions can be reused in the replacement string.
+ *
+ * <p>Examples:
+ *
+ * <ul>
+ *   <li>"*localhost" matches "server.localhost" and captures "server."
+ *   <li>"* localhost" with replacement "localhost *" reorders to "localhost server"
+ *   <li>"* - *" matches "foo - bar" and captures ["foo", " bar"]
+ * </ul>
+ *
+ * @see <a href="https://docs.splunk.com/Documentation/Splunk/latest/SearchReference/Replace">SPL
+ *     replace command</a>
+ */
+public class WildcardReplaceUtils {
+
+  /**
+   * Perform wildcard-based replacement.
+   *
+   * @param input The input string
+   * @param pattern The pattern (may contain wildcards)
+   * @param replacement The replacement (may contain wildcards)
+   * @return The replaced string, or original if no match
+   */
+  public static String replaceWithWildcard(String input, String pattern, String replacement) {
+    if (input == null) {
+      return null;
+    }
+
+    // Fast path: no wildcards = literal replacement
+    if (!pattern.contains("*")) {
+      return input.replace(pattern, replacement);
+    }
+
+    // Match and capture wildcard portions
+    List<String> captures = matchAndCapture(input, pattern);
+    if (captures == null) {
+      // No match - return original
+      return input;
+    }
+
+    // Substitute wildcards in replacement with captured values
+    return substituteWildcards(replacement, captures);
+  }
+
+  /**
+   * Match pattern against input and capture wildcard portions.
+   *
+   * @param input The input string
+   * @param pattern The pattern with wildcards
+   * @return List of captured strings (one per wildcard), or null if no match
+   */
+  public static List<String> matchAndCapture(String input, String pattern) {
```

Comment:
suggestion (if-minor): Compiling an equivalent Regex matcher might be faster and simpler.